decoder/validate.go

```diff
@@ -48,6 +48,28 @@ func (d *PathDecoder) Validate(ctx context.Context) (lang.DiagnosticsMap, error)
 	return diags, nil
 }
 
+// ValidateFile returns a list of Diagnostics for a specific file
+func (d *PathDecoder) ValidateFile(ctx context.Context, filename string) (hcl.Diagnostics, error) {
+	if d.pathCtx.Schema == nil {
+		return hcl.Diagnostics{}, &NoSchemaError{}
+	}
+
+	f, err := d.fileByName(filename)
+	if err != nil {
+		return hcl.Diagnostics{}, err
+	}
+
+	body, ok := f.Body.(*hclsyntax.Body)
+	if !ok {
+		// TODO! error
```

Comment:
👀 

I think we should mimic what we do in other similar methods already - i.e. return error to say that JSON isn't supported.